File: mobile/lib/core/models/correction.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'correction.freezed.dart';
part 'correction.g.dart';

@freezed
abstract class Correction with _$Correction {
  const factory Correction({
    required String id,
    @JsonKey(name: 'venue_id') required String venueId,
    @JsonKey(name: 'suggested_by') required String suggestedBy,
    @JsonKey(name: 'field_name') required String fieldName,
    @JsonKey(name: 'old_value') String? oldValue,
    @JsonKey(name: 'new_value') required String newValue,
    @Default('pending') String status,
    @JsonKey(name: 'reviewed_by') String? reviewedBy,
    @JsonKey(name: 'reviewed_at') DateTime? reviewedAt,
    String? note,
    @JsonKey(name: 'venue_name') String? venueName,
    @JsonKey(name: 'suggested_by_name') String? suggestedByName,
    @JsonKey(name: 'created_at') DateTime? createdAt,
  }) = _Correction;

  factory Correction.fromJson(Map<String, dynamic> json) =>
      _$CorrectionFromJson(json);
}

File: backend/internal/models/correction.go
package models

import "time"

type CorrectionStatus string

const (
	CorrectionStatusPending  CorrectionStatus = "pending"
	CorrectionStatusApproved CorrectionStatus = "approved"
	CorrectionStatusRejected CorrectionStatus = "rejected"
)

type CorrectionSuggestion struct {
	ID          string           `json:"id"`
	VenueID     string           `json:"venue_id"`
	SuggestedBy string           `json:"suggested_by"`
	FieldName   string           `json:"field_name"`
	OldValue    *string          `json:"old_value"`
	NewValue    string           `json:"new_value"`
	Status      CorrectionStatus `json:"status"`
	ReviewedBy  *string          `json:"reviewed_by,omitempty"`
	ReviewedAt  *time.Time       `json:"reviewed_at,omitempty"`
	Note        *string          `json:"note,omitempty"`
	CreatedAt   time.Time        `json:"created_at"`
}
